tsmo/src/parse_schedule_csv_logs.py

```diff
@@ -0,0 +1,110 @@
+""" This is an analysis script for creating plots of entering_times against timestamps of vehicles using scheduling_service logs
+The plots are created for EV and DV states and require a scheduling_service log csv to be passed as an argument """
+
+## How to use this script:
+""" This script needs to be run with an argument for the scheduling_service csv log file. ./parse_schedule_csv_logs.py <log_file.csv>
+A single plot consists of entering times for a run, which is marked by linebreaks in the csv. 
+The resulting plot is between timestamp entry and entering time entry.
+The plot also shows the state ("EV" or "DV") which corresponds with the given data entry. """
+
+### Additional Note
+"""Current implementation adds line breaks when there is no vehicle in the intersection and only has entries when the vehicle has carma engaged.
+This was a useful demarcation for separating runs but the script might need to be modified if implementation changes."""
+
+import csv
+from time import time
+import matplotlib
+import matplotlib.pyplot as plt
+import datetime
+import matplotlib.dates as md 
+import sys
+import pandas as pd
+import math
+import matplotlib.patches as mpatches
+import numpy as np
+
+class Logparser():
+
+    log_file_name_ = ""
+    num_of_runs_ = 0
+    ev_runs_to_count_ = 10
+
+    def __init__(self, file_name):
+        self.log_file_name_ = file_name
+        self.process_runs()
+
+    def convert_to_datetime(self, x):
+        return datetime.datetime.fromtimestamp(int(x)/1000)
+
+
+    def plot_run(self, df):
+        # Method to handle processing of single run stored in pandas dataframe df
+
+        # Separate out values based on states
+        df_ev = df[df["state"] == "EV"]
+        df_dv  = df[df["state"] == "DV"]
+
+
+        # Convert timestamps and entering timestamps to datetime values
+        pd.options.mode.chained_assignment = None  # default='warn'
```

Comment:
I believe you are getting this bc of the SettingWithoutCopy warning right? This is an explanation of why you should use a copy instead of operating on the actual df. https://stackoverflow.com/questions/27673231/why-should-i-make-a-copy-of-a-data-frame-in-pandas

Reply:
Fixed this, thanks!